### PR TITLE
MRG: Allow ASCII BV files

### LINKED ISSUE #3740: AssertionError brainvision.pyc file
Hello,

I can't open my .vhdr files for EEG analysis, there's got to be a pb with my brain vision.pyc file. Can someone guides me to fix this ?
Thxs in advance !
screen shot 2016-11-05 at 19 56 47

### PR DESCRIPTION
**Pull request by Eric89GXL**
Closes #3740.

Still needs tests. Seems to work with data @clementdonde provided.

### CODE REVIEW

File: mne/io/brainvision/brainvision.py [hunk 1]
```diff
@@ -79,30 +80,49 @@ def __init__(self, vhdr_fname, montage=None,
...
-        vhdr_fname = os.path.abspath(vhdr_fname)
         info, fmt, self._order, mrk_fname, montage = _get_vhdr_info(
             vhdr_fname, eog, misc, scale, montage)
         events = _read_vmrk_events(mrk_fname, event_id, response_trig_shift)
         _check_update_montage(info, montage)
         with open(info['filename'], 'rb') as f:
-            f.seek(0, os.SEEK_END)
-            n_samples = f.tell()
-        dtype_bytes = _fmt_byte_dict[fmt]
+            if isinstance(fmt, dict):  # ASCII, this will be slow :(
```
Comment (jona-sassenhagen) [inline mne/io/brainvision/brainvision.py]: *Maybe* (not important) give a warning for this case? ("ASCII detected, loading data will be slow")

File: mne/io/brainvision/brainvision.py [hunk 2]
```diff
@@ -79,30 +80,49 @@ def __init__(self, vhdr_fname, montage=None,
...
+            _read_segments_file(self, data, idx, fi, start, stop, cals, mult,
+                                dtype=dtype, n_channels=n_data_ch,
+                                trigger_ch=self._event_ch)
+        else:
+            offsets = self._raw_extras[fi]
+            with open(self._filenames[fi], 'rb') as fid:
+                fid.seek(offsets[start])
+                block = np.empty((len(self.ch_names), stop - start))
+                for ii in range(stop - start):
+                    line = fid.readline().strip().split().replace(',', '.')
```
Comment (jaeilepp) [inline mne/io/brainvision/brainvision.py]: This should probably be ``line = fid.readline().strip().replace(',', '.').split()``

File: mne/io/brainvision/brainvision.py [hunk 3]
```diff
@@ -318,8 +338,11 @@ def _get_vhdr_info(vhdr_fname, eog, misc, scale, montage):
         Events from the corresponding vmrk file.
     """
     scale = float(scale)
-
+    vhdr_fname = os.path.abspath(vhdr_fname)
     ext = os.path.splitext(vhdr_fname)[-1]
+    if ext == '':  # lazy user gives filename with no extension
+        ext = '.vhdr'
+        vhdr_fname += ext
```
Comment (jaeilepp) [inline mne/io/brainvision/brainvision.py]: Is it not possible to have a file with no extension? Maybe check ``if not op.exists(vhdr_fname)`` before adding the extension?

File: mne/io/brainvision/brainvision.py (hunk 1 above)
Comment (Eric89GXL) [inline mne/io/brainvision/brainvision.py]: I guess we could. I'll let someone add that if they experience bad slowness

**Comment by Eric89GXL** [comment]
Only reason this is WIP is because I don't have a suitable small test file. I could fake one by writing out what we'd expect. Might be the easiest way unless someone has access to BV files and could make a small test file.

**Comment by Eric89GXL** [comment]
Okay I made a test by substituting values. Ready for review/merge from my end.

File: mne/io/brainvision/tests/test_brainvision.py [hunk 4]
```diff
@@ -135,57 +172,49 @@ def test_brainvision_data_lowpass_filters():
...
 
     assert_equal(raw.info['highpass'], 0.1)
     assert_equal(raw.info['lowpass'], 250.)
 
 
 def test_brainvision_data_partially_disabled_hw_filters():
-    """Test reading raw Brain Vision files with heterogeneous amplifier
-       filter settings including non-numeric values
-    """
+    """Test rheterogeneous filter settings including non-numeric values."""
```
Comment (jaeilepp) [inline mne/io/brainvision/tests/test_brainvision.py]: -> heterogeneous

**Comment by Eric89GXL** [comment]
Fixed, thanks @jaeilepp.

@agramfort does it work for you now?

**Comment by agramfort** [comment]
worked !

thx @Eric89GXL

**Pull request merged**